### ISSUE TITLE
"panic: runtime error: invalid memory address or nil pointer dereference" Kill Nomad Cluster

### ISSUE THREAD

**Issue by orarnon** [hashicorp/nomad#4602]
### Nomad version
0.8.4

### Operating system and Environment details
Ubuntu 16.04

### Issue
panic: runtime error: invalid memory address or nil pointer dereference
### Reproduction steps
No idea.
### Nomad Server logs (if appropriate)
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc8 pc=0xf027c3]

goroutine 73 [running]:
github.com/hashicorp/nomad/nomad/structs.(*Node).Ready(...)
    /opt/gopath/src/github.com/hashicorp/nomad/nomad/structs/structs.go:1487
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).findPreferredNode(0xc421f12780, 0x1d7a5c0, 0xc421d97380, 0x8, 0x2bba400, 0xc4203b4200)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:600 +0xd3
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).computePlacements(0xc421f12780, 0x2bb8aa0, 0x0, 0x0, 0xc422687520, 0x1, 0x1, 0xc421d74f18, 0x1)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:448 +0x2ec
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).computeJobAllocs(0xc421f12780, 0xc4231396c0, 0xc420dea0c0)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:410 +0x1586
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).process(0xc421f12780, 0x18, 0xc4226f19a7, 0x7)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:245 +0x471
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).(github.com/hashicorp/nomad/scheduler.process)-fm(0xc4229c7750, 0x0, 0x0)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:144 +0x2a
github.com/hashicorp/nomad/scheduler.retryMax(0x5, 0xc4229c78b0, 0xc4229c78c0, 0xc, 0xffffffffffffffff)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/util.go:271 +0x43
github.com/hashicorp/nomad/scheduler.(*GenericScheduler).Process(0xc421f12780, 0xc421cf86e0, 0xc420202be0, 0x1d7cde0)
    /opt/gopath/src/github.com/hashicorp/nomad/scheduler/generic_sched.go:144 +0x123
github.com/hashicorp/nomad/nomad.(*nomadFSM).reconcileQueuedAllocations(0xc4204f4ba0, 0x205bc4, 0x0, 0x0)
    /opt/gopath/src/github.com/hashicorp/nomad/nomad/fsm.go:1383 +0x8ed
github.com/hashicorp/nomad/nomad.(*nomadFSM).applyReconcileSummaries(0xc4204f4ba0, 0xc422fa3891, 0x8, 0x8, 0x205bc4, 0xc4229c7f76, 0x2b96de0)
    /opt/gopath/src/github.com/hashicorp/nomad/nomad/fsm.go:758 +0x7e
github.com/hashicorp/nomad/nomad.(*nomadFSM).Apply(0xc4204f4ba0, 0xc42070e7e0, 0x2b96de0, 0x3)
    /opt/gopath/src/github.com/hashicorp/nomad/nomad/fsm.go:210 +0x6f1
github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).runFSM.func1(0xc4229807f0)
    /opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/fsm.go:57 +0x15a
github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).runFSM(0xc4201be2c0)
    /opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/fsm.go:120 +0x2fa
github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.runFSM)-fm()
    /opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/api.go:506 +0x2a
github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc4201be2c0, 0xc421236e60)
    /opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/state.go:146 +0x53
created by github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft.(*raftState).goFunc
    /opt/gopath/src/github.com/hashicorp/nomad/vendor/github.com/hashicorp/raft/state.go:144 +0x66
```

**Comment by dadgar** [comment]
Hey thanks for reporting! Will get this fixed for the next release!

**Comment by dadgar** [comment]
Actually already fixed in master! https://github.com/hashicorp/nomad/pull/4474/files

**Comment by orarnon** [comment]
Hi @dadgar and thanks for replying.
Can you elaborate what has caused this issue? It happened when we tried to upload a new job spec but we're not sure that's the reason.
Was this version of master already released? (we have the latest 0.8.4) already
